models/Bid.go

```diff
@@ -168,7 +165,7 @@ func (bid *Bid) Close() error {
 	bid.UpdatedAt = utils.GetCurrentTimeISO8601()
 	bid.Unlock()
 
-	if err := db.Save(bid).Error; err != nil {
+	if err := tx.Save(bid).Error; err != nil {
```

Comment:
There's a bug here (old code). If db gives error, undo the modifications done in the bid. (May not be urgent if the caller doesn't use this object in case of error. Can you once check if the caller is safe?)

Reply:
This method is called from 2 places :- 
1. `CancelOrder` from `User.go` - Here, tx ix rolled back if an error is detected :- 
```go
err = bidOrder.Close(tx)
// don't log if order is already closed
if _, ok := err.(AlreadyClosedError); !ok {
	l.Errorf("Unknown error while saving that bid is cancelled %+v", err)
}
// return the error anyway
if err != nil {
	tx.Rollback()
	return nil, nil, err
}
```

2. `PerformOrderFillTransaction` from `User.go` (This is something @PaiAkshay998 has added in this PR). Here, it is not rolled back in case of an error :- 
```go
if cashLeft < MINIMUM_CASH_LIMIT {
	l.Errorf("Check1: Failed. Not enough cash. Returning reserved cash back to user.")
	bid.Close(tx)
	SaveBidCancelOrderTransaction(bid, biddingUser, tx)
	err := tx.Commit() // Error will still go on to return AskUndone
	l.Errorf("Error while commiting bidClose. Error: %+v", err)
	go updateDataStreams(nil, nil, nil, nil)
	go SendNotification(biddingUser.Id, fmt.Sprintf("Your Buy order#%d has been closed due to insufficient cash", bid.Id), false)
	return AskUndone, BidDone, nil
}
```